storage/memory/peer_store.go

```diff
@@ -119,13 +126,36 @@ func New(cfg Config) (storage.PeerStore, error) {
 		}
 	}()
 
+	ps.wg.Add(1)
+	go func() {
+		defer ps.wg.Done()
+		t := time.NewTicker(cfg.PrometheusReportingInterval)
+		for {
+			select {
+			case <-ps.closing:
+				t.Stop()
+				select {
+				case <-t.C:
```

Comment:
Hmm, the idea was to drain the channel so it can be GC'ed, but apparently there's a more official way to stop a timer: https://golang.org/pkg/time/#Timer.Stop
Will change the other goroutine too